### PR TITLE
Fix timing bug in Mongodb replicaset configuration

### LINKED ISSUE #17: Error while adding a new replicaset member
When configuring a node which is not yet member of a replicaset, different errors can occur.

Depending on whether the primary was configured first or not.

Primary first:
`ERROR: configuring replicaset returned: #<BSON::OrderedHash:0x3f81e4b9a140 {"ok"=>1.0}>`
In this case I _think_ that the operation is successful.

New member first:
`WARN: Could not connect to database: 'localhost:27017'`
and the new member is not added to the primary (possible iptables not yet configured at the primary?)

I think that the code responsible for adding replicaset members should be refactored to take better into account the different situations.

### PR DESCRIPTION
**Pull request by danieltahara**
Since start_server() and restart_server() do not actually verify whether the
port is open and receiving connections (even though mongo starts up,
pre-alloc/other setup takes a few minutes), the replicaset recipe was failing to
establish a connection with the db.  The change to the initd file waits for the
port to be open before proceeding, ensuring that the replicaset recipe will
work.

Fixes #17

### CODE REVIEW

File: templates/default/mongodb.init.erb [hunk 1]
```diff
@@ -92,7 +92,12 @@ running() {
     [ ! -f "$PIDFILE" ] && return 1
     pid=`cat $PIDFILE`
     running_pid $pid $DAEMON || return 1
-    return 0
+    for i in `seq 1 20`; do
+      echo -n "."
+      sleep 15
+      nc -z localhost 27017 && return 0
```
Comment (dcrosta) [inline templates/default/mongodb.init.erb]: This should use `<%= node['mongodb']['port'] %>` rather than hard-coding 27017

Comment (dcrosta) [inline templates/default/mongodb.init.erb]: Also, we should test first and then sleep, so that each `service mongodb restart` doesn't take a minimum of 15 seconds (after the initial allocation, starting MongoDB should take <1s, in my experience).

Comment (danieltahara) [inline templates/default/mongodb.init.erb]: Good points.  Done!

**Comment by ghost** [comment]
Anyone have fixed this for the centos init file?

**Comment by josephholsten** [comment]
Could you please apply this patch upstream at https://github.com/mongodb/mongo/blob/master/debian/init.d?

**Comment by josephholsten** [comment]
I think this looks fine, it just needs to be rebased onto master.

**Comment by jamesonjlee** [comment]
until we figure out a better way, 15 second delay it is, up to 3 minutes..
